### PR TITLE
[FLINK-27834] Copy all files except those filtered by .dockerignore

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

### CODE REVIEW

**Comment by tisonkun** [comment]
cc @morhidi @SteNicholas

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+*.jar
+*.war
+*.nar
+*.ear
+*.zip
+*.tar.gz
+*.rar
+
+### Maven template
+target/
```
Comment (wangyang0918) [inline .dockerignore:20]: Could this line exclude all the target directories of submodules?

Comment (wangyang0918) [reply .dockerignore:20]: It seems that the `.dockerignore` is a little different from `.gitignore`. https://docs.docker.com/engine/reference/builder/#dockerignore-file

Maybe we need to use `**/target`, `**/*.iml`, etc.

Comment (tisonkun) [reply .dockerignore:20]: You're right. Will update.

Comment (tisonkun) [reply .dockerignore]: Updated 97fab7e

**Review by gyfora — APPROVED** [review]

**Comment by tisonkun** [comment]
Thanks for your reviews!

**Pull request merged**